### ISSUE TITLE
Failing to compile extensions on MacOS

### ISSUE THREAD

**Issue by pavyamsiri** [pynbody/pynbody#848]
**Describe the bug**
I am trying to build pynbody from source as part of installing it into my venv on my MacOS M3 laptop. I am running into compilation issues however that necessitates that I modify the setup.py script. The first issue seems to be that the default clang does not recognise the -fopenmp flag.

**To Reproduce**
Just running the setup.py script on a fresh venv with the dependencies for pynbody installed leads to the compilation issue.

**Expected behaviour or result**
Please provide a clear and concise description of what you expected, and how it differs from the result.

**Setup (please complete the following information):**
- OS: MacOS Sonoma 14.6.1 (23G93)
 - Python version: 3.12.6
 - Numpy version: 2.1.1
 - Pynbody version: master branch commit (d647ebab8fd239b00dde624acd1c5f2d1dc2bcc3) ~v2.0.0-beta.13

**Additional context**
I have tried to get it to compile myself and I have succeeded but it requires knowing certain paths so I don't have a PR ready as it is not generalised.

Here are the steps I did to get it to work:
1. To use -fopenmp you need to have the flag `-Xpreprocessor` right before so that the flag doesn't get rejected by the clang front end.
2. Add the path to the directory containing the libomp library like `-L/path/to/lib/` followed by `-lomp`

I also ran into some issues with the C++ standard being too old. I think the clang compiler's default standard is older than C++14 used here so I had to add `language=c++` to each `Extension` along with adding `std=c++14` to the extra_link_args variable.

Hopefully that is helpful. I might also try to fix some of the issues with my branch to see if it can generalised.

**Comment by pavyamsiri** [comment]
I think another factor is that omp is not supported by clang so you have to download it separately and also include and link it as well. GCC has OpenMP support so you don't need to do anything extra.

**Comment by apontzen** [comment]
Here are the current recommendations

https://pynbody.readthedocs.io/latest/installation.html#macos-compilers

Note that it doesn't even mention trying to force openmp compilation with apple clang because this can lead to all sorts of downstream problems.

If anything could be usefully clarified there please suggest!

**Comment by pavyamsiri** [comment]
Ah I should've read the documentation first! I saw a comment mentioning that in setup.py but I got it compiled so I didn't think much of it.

Yeah it seems like adding support for Apple clang might be too much of a hassle and prone to bugs.

I'm not sure if there needs to be any further clarification although I think Apple clang does have openmp support but it is not official? I'm not entirely certain but I got it to work kind of but I ran into a few crashes but I'm not sure if that is due to it being the latest commit on master.

Thanks for the link to the documentation. I think I will close the issue because the current workaround is fine.
